lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Random.hs

```diff
@@ -112,64 +114,89 @@ random
     :: forall m e. MonadRandom m
     => CoinSelectionOptions e
     -> NonEmpty TxOut
+    -> Quantity "lovelace" Word64
     -> UTxO
     -> ExceptT (ErrCoinSelection e) m (CoinSelection, UTxO)
-random opt outs utxo = do
+random opt outs (Quantity totalWithdraw) utxo = do
     let descending = NE.toList . NE.sortBy (flip $ comparing coin)
     let nOuts = fromIntegral $ NE.length outs
     let maxN = fromIntegral $ maximumNumberOfInputs opt nOuts
+    let totalOut = sum (getCoin . coin <$> outs)
     randomMaybe <- lift $ runMaybeT $
-        foldM makeSelection (maxN, utxo, []) (descending outs)
+        foldM
+            (\acc out ->
+                let
+                    withdraw = totalWithdraw
+                        `proportionallyTo` (getCoin (coin out) % totalOut)
```

Comment:
So here, we are guaranteed that the ratio given as second argument is always `<= 1`

Reply:
real question, maybe trivial... but why do we need to use `proportionallyTo` at all? why this rule to transfer upon withdrawing proportionally to relative weight in total outs?  Why not to transfer to the biggest output? Or smallest output? or create new out?